src/Actions/OutputsDslQueryAction.php

```diff
@@ -0,0 +1,99 @@
+use Rapidez\StatamicQueryBuilder\Parsers\DSL\IsNotNullParser;
+use Rapidez\StatamicQueryBuilder\Parsers\DSL\IsNullParser;
+use Rapidez\StatamicQueryBuilder\Parsers\DSL\LessThanOrEqualParser;
+use Rapidez\StatamicQueryBuilder\Parsers\DSL\LessThanParser;
+use Rapidez\StatamicQueryBuilder\Parsers\DSL\LikeParser;
+use Rapidez\StatamicQueryBuilder\Parsers\DSL\NotBetweenParser;
+use Rapidez\StatamicQueryBuilder\Parsers\DSL\NotInParser;
+use Rapidez\StatamicQueryBuilder\Parsers\DSL\NotLikeParser;
+use Rapidez\StatamicQueryBuilder\Parsers\DSL\NotTermParser;
+use Rapidez\StatamicQueryBuilder\Parsers\DSL\StartsWithParser;
+use Rapidez\StatamicQueryBuilder\Parsers\DSL\TermParser;
+
+class OutputsDslQueryAction
+{
+    protected array $operators = [
+        '=' => TermParser::class,
+        '!=' => NotTermParser::class,
+        'LIKE' => LikeParser::class,
+        'NOT LIKE' => NotLikeParser::class,
+        'STARTS_WITH' => StartsWithParser::class,
+        'ENDS_WITH' => EndsWithParser::class,
+        'IN' => InParser::class,
+        'NOT IN' => NotInParser::class,
+        '>' => GreaterThanParser::class,
+        '<' => LessThanParser::class,
+        '>=' => GreaterThanOrEqualParser::class,
+        '<=' => LessThanOrEqualParser::class,
+        'BETWEEN' => BetweenParser::class,
+        'NOT_BETWEEN' => NotBetweenParser::class,
+        'IS_NULL' => IsNullParser::class,
+        'IS_NOT_NULL' => IsNotNullParser::class,
+        'LAST_X_DAYS' => LastXDaysParser::class,
+        'NEXT_X_DAYS' => NextXDaysParser::class,
+        'THIS_WEEK' => ThisWeekParser::class,
+        'THIS_MONTH' => ThisMonthParser::class,
+    ];
+
+    public function build(array $config): array
+    {
+        $groupConjunction = strtoupper($config['globalConjunction'] ?? 'AND');
+        $globalKey = $groupConjunction === 'OR' ? 'should' : 'must';
+        $groups = $config['groups'] ?? [];
+        $limit = (int) ($config['limit'] ?? 10);
+
+        $clauses = [];
+
+        foreach ($groups as $group) {
+            $groupConjunction = strtoupper($group['conjunction'] ?? 'AND');
+            $groupKey = $groupConjunction === 'OR' ? 'should' : 'must';
+            $conditions = [];
+
+            foreach ($group['conditions'] as $condition) {
+                $conditions[] = $this->mapCondition($condition);
+            }
+
+            if (count($groups) === 1 && $groupKey === $globalKey) {
+                $clauses = array_merge($clauses, $conditions);
+            } else {
+                $clauses[] = ['bool' => [$groupKey => $conditions]];
+            }
+        }
+
+        return [
+            'query' => ['bool' => [$globalKey => $clauses]],
+            'size' => $limit,
+            'from' => 0,
+        ];
+    }
+
+    private function mapCondition(array $condition): array
+    {
+        $operator = strtoupper($condition['operator']);
+        $field = is_numeric($condition['attribute']) ? $condition['attribute'] : $condition['attribute'].'.keyword';
```

Comment:
I could see problems arising when an indexed attribute is not text, such as price.
keyword is only available on text fields, perhaps we should fetch, cache and determine the mapping from ES
https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-indices-get-mapping-1

Reply:
I don't think this is the way either, you'd want to check it on `$condition['value']` instead of attribute.

<img width="324" alt="Screenshot 2025-06-30 at 15 25 41" src="https://github.com/user-attachments/assets/d327c346-0c67-4256-a27d-926e7f8fb9a5" />

Reply:
Let's get it from elastic, will discuss this tomorrow